core/shared/src/main/scala/spire/math/Interval.scala

```diff
@@ -767,6 +767,48 @@ sealed abstract class Interval[A](implicit order: Order[A]) extends Serializable
 
   def foldOver[B](init: B, step: A)(f: (B, A) => B)(implicit ev: AdditiveMonoid[A], nt: NumberTag[A]): B =
     iterator(step).foldLeft(init)(f)
+
+  /**
+    * Result of overlapping this interval with another one.
+    * Can be one of the following:
+    * - [[spire.math.interval.Equals]] if intervals are equal
+    * - [[spire.math.interval.StrictlyLess]] if intervals are notEmpty don't intersect
+    * - [[spire.math.interval.LessAndOverlaps]] if intervals intersect and neither is a subset of another
```

Comment:
Similar here - wonder if we could rename to `Overlaps` or `PartialOverlap` or something.